File: system-test/dotnet/Dsl.Core/Scenario/ExecutionResultContext.cs
namespace Dsl.Core.Scenario;

/// <summary>
/// Context extracted from execution result - order number and coupon code from the executed operation.
/// </summary>
public class ExecutionResultContext
{
    private static readonly ExecutionResultContext _empty = new(null, null);

    public ExecutionResultContext(string? orderNumber, string? couponCode)
    {
        OrderNumber = orderNumber;
        CouponCode = couponCode;
    }

    public static ExecutionResultContext Empty() => _empty;

    public string? OrderNumber { get; }
    public string? CouponCode { get; }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/core/scenario/ExecutionResultContext.java
package com.optivem.shop.dsl.core.scenario;

public class ExecutionResultContext {
    private static final ExecutionResultContext EMPTY = new ExecutionResultContext(null);

    private final String orderNumber;

    public ExecutionResultContext(String orderNumber) {
        this.orderNumber = orderNumber;
    }

    public static ExecutionResultContext empty() {
        return EMPTY;
    }

    public String getOrderNumber() {
        return orderNumber;
    }
}

